docs/protocol.md

```diff
@@ -662,7 +673,235 @@ Response (in case of success)
+                "message_ids": [
+                    "z6SbjJ0Hw36k8L09-GVRq4PNmi06yQX4e8aZRSbUDwc=",
+                    "txbTmVMwCDkZdoaAiEYfAKozVizZzkeMkeOlzq5qMlg="
+                ]
+            }
+        ]
+    }
+}
+
+```
+
+The heartbeat can then trigger the internal logic of a server: when it receives message ids that it has not already delivered it will execute a `get_messages_by_id` RPC. However, this is just a consequence of that specific condition, normally the heartbeat is just a way to communicate to others that the sender is alive, thus it doesn't expect any answer back.
+
+<details>
+<summary>
+💡 See the full specification
+</summary>
+  
+```json5
+// ../protocol/query/method/heartbeat.json
+
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/dedis/popstellar/master/protocol/query/method/heartbeat.json",
+    "description": "Heartbeat message",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+        "method": {
+            "description": "[String] operation to be performed by the query",
+            "const": "heartbeat"
+        },
+        "params": {
+            "description": "[Array] of objects containing each a channel and the list of ids from that channel",
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "message_ids_by_channel_id": {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "additionalProperties": false,
+                        "properties": {
+                            "channel": {
+                                "description": "[String] name of the channel",
+                                "type": "string",
+                                "pattern": "^/root(/[^/]+)*$"
+                            },
+                            "message_ids": {
+                                "description": "[Array] of message_ids corresponding to that channel",
+                                "type": "array",
+                                "items": {
+                                    "type": "string"
+                                }
+                            }
+                        },
+                        "required": ["channel", "message_ids"]
+                    }
+                }
+            },
+            "required": ["message_ids_by_channel_id"]
+        },
+        "jsonrpc": {
+            "$comment": "Defined by the parent, but needed here for the validation"
+        },
+
+        "id": {
+            "type": "integer"
+        }
+    },
+    "required": ["method", "params", "id", "jsonrpc"]
+}
+
+```
+</details>
+
+## Retrieving messages from server using ids 
+
+🧭 **RPC Message** > **Query** > **Get messages by their ids**
+  
+The purpose of this RPC is to request missed messages so far.
```

Comment:
From a protocol perspective, try to define first and foremost what the RPC does and allow, not how it is used. It's the property of any API that eventually it'll get used for something vastly different 😁 In this case, we can easily anticipate that eventually the clients might use this.

A suggested better description:

```suggestion
The purpose of this RPC is to selectively retrieve multiple messages by their ID. This allows the caller to only retrieve the messages it doesn't already have.
```